test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperInputExtensionsTest.cs

```diff
@@ -0,0 +1,324 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Mvc.TestCommon;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Core
+{
+    /// <summary>
+    /// Test the <see cref="HtmlHelperInputExtensions" /> class.
+    /// </summary>
+    public class HtmlHelperInputExtensionsTest
+    {
+        [Fact]
+        public void CheckboxHelpers_UsesSpecifiedExpression()
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper();
+
+            // Act
+            var checkboxResult = helper.CheckBox("Property1");
+
+            // Assert
+            Assert.Equal(
+                "<input id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[checkbox]]\" value=\"HtmlEncode[[true]]\" />" +
+                "<input name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[hidden]]\" value=\"HtmlEncode[[false]]\" />",
+                HtmlContentUtilities.HtmlContentToString(checkboxResult));
```

Comment:
I'm agreeing on the value of testing to confirm the generated HTML. But I would rather that be done in tests that call `IHtmlHelper` methods directly. Here it just gets redundant.